src/vs/workbench/electron-browser/workbench.ts

```diff
@@ -228,9 +228,14 @@ export class Workbench implements IPartService {
 				viewletId = this.storageService.get(SidebarPart.activeViewletSettingsKey, StorageScope.WORKSPACE, viewletId); // help developers and restore last view
 			}
 
-			if (!this.sideBarHidden && !!viewletId) {
-				const viewletTimerEvent = timer.start(timer.Topic.STARTUP, strings.format('Opening Viewlet: {0}', viewletId));
-				compositeAndEditorPromises.push(this.sidebarPart.openViewlet(viewletId, false).then(() => viewletTimerEvent.stop()));
+			// If external viewlet is the last active viewlet, defer its construction until all extensions are loaded
+			if (!viewletRegistry.getViewlet(viewletId)) {
```

Comment:
@octref this is ugly and should be removed. instead the viewlet service should be fit enough to handle both extension viewlets and normal viewlets. if you clean that up you will not need this if-else condition at all 👍